thrift/types.go

```diff
@@ -115,11 +115,15 @@ func parseBinaryList(vl []interface{}) ([]byte, error) {
 	bs := make([]byte, 0, len(vl))
 	for _, v := range vl {
 		switch v := v.(type) {
+		// YAML uses int for all small values. For large values, it may use int64 or uint64
+		// but those are not valid bool values anyway, so we don't need to check them.
 		case int:
-			// YAML uses int for all small values. For large values, it may use int64 or uint64
-			// but those are not valid bool values anyway, so we don't need to check them.
-			if v < 0 || v >= (1<<8) {
-				return nil, fmt.Errorf("failed to parse list of bytes: %v is not a byte", v)
+			// from https://thrift.apache.org/docs/types, byte is an 8-bit signed integer
+			// For Go, thriftrw uses a Go byte, which is an uint8 so it is possible to have 0 to 225 over the wire.
+			// Apache Java clients respects the signed int8 and send -128 to 127.
+			// To make both sides happy, check between -128 and 225
+			if v < -(1<<7) || v >= (1<<8) {
```

Comment:
can we use math.MinInt8 and math.MaxUint8 for min and max values? I believe that is more cleaner

Reply:
given we are talking in explicit number range, so I think we should avoid indirection by using math util. being more explicit here IMO is cleaner.

Reply:
+1